src/utils/index.ts

```diff
@@ -8,3 +11,57 @@ export const patchPnpmEnv = (inputs: Inputs): NodeJS.ProcessEnv => ({
   ...process.env,
   PATH: getBinDest(inputs) + path.delimiter + process.env.PATH,
 })
+
+/**
+ * Reads and parses a potential .npmrc file to extract registry configuration
+ * @param workspaceRoot - The root directory of the workspace
+ * @returns The registry URL if found, otherwise undefined
+ */
+export async function readNpmrcRegistry(workspaceRoot: string): Promise<string | undefined> {
+  try {
+    const npmrcPath = path.join(workspaceRoot, '.npmrc');
+
+    debug(`Attempting to read .npmrc file from: ${npmrcPath}`);
+    const content = await readFile(npmrcPath, 'utf8');
+
+    for (const line of content.split('\n')) {
+      const trimmedLine = line.trim();
+
+      // skip comments and empty lines
+      if (!trimmedLine || /^[#;]/.test(trimmedLine)) {
```

Comment:
We should minimize the use of regex.

```suggestion
      if (!trimmedLine || trimmedLine.startsWith('#') || trimmedLine.startsWith(';')) {
```

Of course, this whole `if continue` block should be a `filter` in the above suggestion.